client/src/app/slides/motions/motion/motion-slide.component.ts

```diff
@@ -324,70 +324,79 @@ export class MotionSlideComponent extends BaseMotionSlideComponent<MotionSlideDa
         // const changes: ViewUnifiedChange[] = Object.assign([], this.allChangingObjects);
         const motion = this.data.data;
 
-        switch (this.crMode) {
-            case ChangeRecoMode.Original:
-                return this.lineNumbering.insertLineNumbers(motion.text, this.lineLength, this.highlightedLine);
-            case ChangeRecoMode.Changed:
-                return this.diff.getTextWithChanges(
-                    motion.text,
-                    this.getAllTextChangingObjects(),
-                    this.lineLength,
-                    this.highlightedLine
-                );
-            case ChangeRecoMode.Diff:
-                let text = '';
-                const changes = this.getAllTextChangingObjects().filter(change => {
-                    return change.showInDiffView();
-                });
-                changes.forEach((change: ViewUnifiedChange, idx: number) => {
-                    if (idx === 0) {
-                        const lineRange = { from: 1, to: change.getLineFrom() };
-                        text += this.extractMotionLineRange(motion.text, lineRange, true, this.lineLength);
-                    } else if (changes[idx - 1].getLineTo() < change.getLineFrom()) {
-                        const lineRange = {
-                            from: changes[idx - 1].getLineTo(),
-                            to: change.getLineFrom()
-                        };
-                        text += this.extractMotionLineRange(motion.text, lineRange, true, this.lineLength);
-                    }
-                    text += this.diff.getChangeDiff(motion.text, change, this.lineLength, this.highlightedLine);
-                });
-                text += this.diff.getTextRemainderAfterLastChange(
-                    motion.text,
-                    changes,
-                    this.lineLength,
-                    this.highlightedLine
-                );
-                return text;
-            case ChangeRecoMode.Final:
-                const appliedChanges: ViewUnifiedChange[] = this.getAllTextChangingObjects().filter(change =>
-                    change.showInFinalView()
-                );
-                return this.diff.getTextWithChanges(motion.text, appliedChanges, this.lineLength, this.highlightedLine);
-            case ChangeRecoMode.ModifiedFinal:
-                if (motion.modified_final_version) {
-                    return this.lineNumbering.insertLineNumbers(
-                        motion.modified_final_version,
+        if (!!motion.text) {
```

Comment:
Please use a fast-return here:
```
if (!motion.text) {
    return null;
}
<rest of method>
```
This reduces the complexity, becuase base-cases are quickly sorted out.